lib/amdResolve.js

```diff
@@ -90,10 +91,20 @@ module.exports = function resolve(pattern, {
     const fileTypePattern  = new RegExp(`\\.${fileExtension}$`);
     const extensionPattern = new RegExp(`^${targetExtension}/`);
 
+    //check if the pattern has an alias
+    const matchinAliases =  Object.entries(aliases)
+        .filter( ([alias]) => new RegExp(`^${alias}/`).test(pattern) );
+
+    //resolve the aliases, only to resolve the modules against the file system
+    const aliasedPattern = matchinAliases
+        .reduce( (acc, [alias, replacement]) => {
+            return acc.replace(new RegExp(`^${alias}`), replacement);
```

Comment:
After the first alias replace, non of the others will match, so make no sense to have `matchinAliases`. You should find the first that matches and replace only that.